### DISCUSSION TITLE
Conductor with mysql example reference

### DISCUSSION THREAD

**Discussion by chandruibm** [Netflix/conductor-community#89] [Q&A]
Is there a working example of conductor using mysql or any other persistence layer other than dynomite & memory. Document seems very poor on conductor extension. It would be good if there are any step by step information on how to configure conductor extension and an example

**Comment by apanicker-nflx** [comment]
Please follow the integration tests setup in the test-harness module for reference, for the time being. We will be updating the documentation to better reflect this setup soon. 
If you have any questions regarding specific configurations, our community members who actively use MySQL as persistence layer would be able to help out.

**Comment by maheshyaddanapudi** [comment]
Hi,

Please have a look at docker folder in this repository https://github.com/maheshyaddanapudi/conductor/tree/all

It has options for MySQL, Postgres, Oracle and Mongo as persistence options.
All are tested and shown in the integration tests, they all pass as expected.

I have more robustly tested Oracle and MySQL.

Hope that helps.

Thanks and Regards,
Mahi.

**Reply by philipsamuel** [reply]
Hi @maheshyaddanapudi  - i tried using your repo as mentioned above , when using mysql, it is failing with dependency errors.

The config i changed was like below to enable mysql . Facing similar issues while using conductor-community as well. 

#conductor.db.type=memory

conductor.db.type=mysql

#Cache expiry for the task definitions in seconds
conductor.mysql.taskDefCacheRefreshInterval=60

#use spring datasource properties to configure MySQL connection
#Create the DB before starting
spring.datasource.url=jdbc:mysql://localhost:3306/conductor
spring.datasource.username=username
spring.datasource.password=password
spring.datasource.hikari.maximum-pool-size=10
spring.datasource.hikari.auto-commit=true



Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
5317 [main] ERROR org.springframework.boot.SpringApplication [] - Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkResource' defined in URL [jar:file:<PATH-MASKED>/conductor/rest/build/libs/conductor-rest-3.0.0-SNAPSHOT.jar!/com/netflix/conductor/rest/controllers/WorkflowBulkResource.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkServiceImpl' defined in URL [jar:file:<PATH-MASKED>/conductor/core/build/libs/conductor-core-3.0.0-SNAPSHOT.jar!/com/netflix/conductor/service/WorkflowBulkServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowExecutor' defined in URL [jar:file:<PATH-MASKED>/conductor/core/build/libs/conductor-core-3.0.0-SNAPSHOT.jar!/com/netflix/conductor/core/execution/WorkflowExecutor.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'deciderService' defined in URL [jar:file:<PATH-MASKED>/conductor/core/build/libs/conductor-core-3.0.0-SNAPSHOT.jar!/com/netflix/conductor/core/execution/DeciderService.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mySqlMetadataDAO' defined in class path resource [com/netflix/conductor/mysql/config/MySQLConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.netflix.conductor.mysql.dao.MySQLMetadataDAO]: Factory method 'mySqlMetadataDAO' threw exception; nested exception is java.lang.IllegalArgumentException
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:799) ~[conductor-es7-persistence-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:228) ~[conductor-es7-persistence-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1361) ~[conductor-es7-persistence-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1208) ~[conductor-es7-persistence-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]

**Comment by BhuhanJ** [comment]
Hi @philipsamuel 
Have you found any solution ? I am facing the same issue while integrating with mysql.
FYI - I am not using docker.

Any help will be appreciated.
